### PR TITLE
Implement ISDDNET and Time Score cleanly

### PR DESCRIPTION
**Pull request by Learath2**
This is an attempt at implementing the two things in a cleaner way.

Deprecated #1320 
Willfix #1622  soon(tm)

### CODE REVIEW

File: src/game/client/components/scoreboard.cpp [hunk 1]
```diff
@@ -294,7 +294,7 @@ void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const ch
 	// render headlines
 	y += 50.0f;
 	float HeadlineFontsize = 22.0f;
-	const char* pScore = (m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard && m_pClient->m_AllowTimeScore[g_Config.m_ClDummy]) ? Localize("Time") : Localize("Score");
+	const char* pScore = (m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard) ? Localize("Time") : Localize("Score");
```
Comment (12pm) [inline src/game/client/components/scoreboard.cpp]:
```suggestion
	const char *pScore = (m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard) ? Localize("Time") : Localize("Score");
```

**Comment by Learath2** [comment]
I think the time-score thing itself is a ill-conceived feature.

I kinda went overboard with it's customisability, it could also be a flag but I don't think we need this in the codebase :/

**Comment by Learath2** [comment]
This I'm really not sure if it's the best solution but atleast it's clean and easy to move around.
`IsDDNet` in an ideal world would disappear into a couple flags in `GameInfoEx`

**Comment by Learath2** [comment]
Also don't merge it like this, I'd rather rebase and squash this beforehand

**Review by def- — APPROVED** [review]
lgtm, but i'm not too familiar with this code.

**Comment by fokkonaut** [comment]
Its good

File: src/game/client/components/hud.cpp [hunk 2]
```diff
@@ -340,7 +340,7 @@ void CHud::RenderScoreHud()
 				{
 					CServerInfo Info;
 					Client()->GetServerInfo(&Info);
-					if(IsRace(&Info) && g_Config.m_ClDDRaceScoreBoard && m_pClient->m_AllowTimeScore[g_Config.m_ClDummy])
+					if(m_pClient->TimeScore() && g_Config.m_ClDDRaceScoreBoard)
```
Comment (Ryozuki) [inline src/game/client/components/hud.cpp]: Don't mind me but, isn't it faster to put `g_Config.m_ClDDRaceScoreBoard` on the left side of the && so you can avoid unnecessari function calls to timescore

File: src/game/client/components/scoreboard.cpp [hunk 3]
```diff
@@ -231,7 +231,7 @@ void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const ch
 		}
 	}
 
-	if(m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard && m_pClient->m_AllowTimeScore[g_Config.m_ClDummy])
+	if(m_pClient->TimeScore() && g_Config.m_ClDDRaceScoreBoard)
```
Comment (Ryozuki) [inline src/game/client/components/scoreboard.cpp]: same here

File: src/game/client/components/scoreboard.cpp [hunk 4]
```diff
@@ -294,7 +294,7 @@ void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const ch
 	// render headlines
 	y += 50.0f;
 	float HeadlineFontsize = 22.0f;
-	const char* pScore = (m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard && m_pClient->m_AllowTimeScore[g_Config.m_ClDummy]) ? Localize("Time") : Localize("Score");
+	const char *pScore = (m_pClient->TimeScore() && g_Config.m_ClDDRaceScoreBoard) ? Localize("Time") : Localize("Score");
```
Comment (Ryozuki) [inline src/game/client/components/scoreboard.cpp]: :)

File: src/game/client/components/scoreboard.cpp [hunk 5]
```diff
@@ -414,7 +414,7 @@ void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const ch
 		}
 
 		// score
-		if(m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard && m_pClient->m_AllowTimeScore[g_Config.m_ClDummy])
+		if(m_pClient->TimeScore() && g_Config.m_ClDDRaceScoreBoard)
```
Comment (Ryozuki) [inline src/game/client/components/scoreboard.cpp]: :o

**Pull request merged**